datastore/src/main/java/io/spine/server/storage/datastore/DsAggregateStorage.java

```diff
@@ -124,24 +140,38 @@ public void writeEventCountAfterLastSnapshot(I id, int eventCount) {
                                                      .build());
     }
 
+    @SuppressWarnings("EnumSwitchStatementWhichMissesCases") // Only valuable cases.
     @Override
     protected void writeRecord(I id, AggregateEventRecord record) {
         checkNotNull(id);
 
         final String stringId = Stringifiers.toString(id);
-        String eventId = Stringifiers.toString(record.getEvent()
-                                                     .getId());
-        if (eventId.isEmpty()) {
-            // Snapshots have no Event IDs.
-            eventId = SNAPSHOT + stringId;
+        final String recordId;
+        final Version version;
+        final KindCase kind = record.getKindCase();
+        switch (kind) {
+            case EVENT:
+                recordId = Stringifiers.toString(record.getEvent().getId());
+                version = record.getEvent()
+                                .getContext()
+                                .getVersion();
+                break;
+            case SNAPSHOT:
+                recordId = SNAPSHOT + stringId;
+                version = record.getSnapshot()
+                                .getVersion();
+                break;
+            default:
+                throw new IllegalArgumentException(record.getKindCase()
```

Comment:
Arguably, it looks like a case for `IllegalStateException` because _this_ part of code cannot handle an `AggregateEventRecord`. Anyway, we need a message which explains the problem in more details than just a name of the kind case.